libraries/abi_generator/abi_generator.cpp

```diff
@@ -477,12 +494,25 @@ clang::QualType abi_generator::get_vector_element_type(const clang::QualType& qt
   return arg0.getAsType();
 }
 
-string abi_generator::get_vector_element_type(const string& type_name) {
+string abi_generator::get_vector_array_element_type(const string& type_name) {
   if( is_vector(type_name) )
     return type_name.substr(0, type_name.size()-2);
+  if( is_fixed_array(type_name) ) {
+     auto p = type_name.find_first_of('[');
+     if (p != string::npos) {
+        while (p > 0 && type_name[p] == ' ') --p;
+        return type_name.substr(0, p);
+     }
+  }
   return type_name;
 }
 
+clang::QualType abi_generator::get_array_element_type(const clang::QualType& qt) {
```

Comment:
I don't see this code used in this code you have added.  Did something not get merged in, or was this method not needed?  If not needed remove.  If not merged, does it need to be, or is that part of a separate PR?

Reply:
i see. will remove this function.